plugins/home/src/homePageComponents/StarredEntities/Content.tsx

```diff
@@ -32,29 +34,66 @@ import {
 } from '@material-ui/core';
 import StarIcon from '@material-ui/icons/Star';
 import React from 'react';
+import useAsync from 'react-use/lib/useAsync';
 
 /**
  * A component to display a list of starred entities for the user.
  *
  * @public
  */
 export const Content = () => {
+  const catalogApi = useApi(catalogApiRef);
   const catalogEntityRoute = useRouteRef(entityRouteRef);
   const { starredEntities, toggleStarredEntity } = useStarredEntities();
 
+  // Grab starred entities from catalog to ensure they still exist and also retrieve display titles
+  const entities = useAsync(async () => {
+    if (!starredEntities.size) {
+      return [];
+    }
+
+    const filter = [...starredEntities]
+      .map(ent => parseEntityRef(ent))
+      .map(ref => ({
+        kind: ref.kind,
+        'metadata.namespace': ref.namespace,
+        'metadata.name': ref.name,
+      }));
+
+    return (
+      await catalogApi.getEntities({
+        filter,
+        fields: [
+          'kind',
+          'metadata.namespace',
+          'metadata.name',
+          'metadata.title',
+        ],
+      })
+    ).items;
+  }, [catalogApi, starredEntities]);
+
   if (starredEntities.size === 0)
     return (
       <Typography variant="body1">
         You do not have any starred entities yet!
       </Typography>
     );
 
-  return (
+  if (entities.loading) {
+    return <Progress />;
+  }
+
+  return entities.error ? (
+    <Typography>Error loading entities: {entities.error}</Typography>
+  ) : (
     <List>
-      {Array.from(starredEntities).map(entity => (
-        <ListItem key={entity}>
-          <Link to={catalogEntityRoute(parseEntityRef(entity))}>
-            <ListItemText primary={parseEntityRef(entity).name} />
+      {entities.value?.map(entity => (
```

Comment:
If you iterate the returned entities instead of the starred, you mess up the ordering, which I think starred should be responsible for. The async function above should probably return `new Set(response.items.map(stringifyEntityRef))` instead, and then starred items checked against that set

Reply:
hmm how important is it really to maintain some sort of ordering with starred entities? considering this is a mix of any entity kind you have starred (components, apis, systems, etc). 

Stringifying the entity response would lose the `title` field which was the other motivation behind this change. We _could_ create a map of `{ [entityRef]: Entity }` to check against `starredEntities` but there's also a concern of ref mismatches ie. the ref strings in `starredEntities` may not be fully qualified (kind:namspace/name) https://github.com/backstage/backstage/blob/02e35fe7688c47e49b785ad4ce643c1bf0ae86bc/plugins/catalog-react/src/hooks/useStarredEntities.ts#L37
A remedy for this could be to "normalize" the refs in `starredEntities` via `stringifyEntityRef(parseEntityRef(ref))` but again I'm not sure ordering is that important to warrant the additional complexity here to achieve that. 

Let me know if I'm over-complicated things here 😅  I'm ok with making the change if this is what we really want

Reply:
It's just that this surely introduces chaos. The ordering of results from the catalog is undefined and is directly connected to the rendered order here right? I may be missing something, and I've left the computer for now :)

Reply:
Went ahead and updated to sort alphabetically, let me know what you think!